### PR TITLE
Prevent crashing when the retry key is nil (redis-rb >= 5)

### PR DESCRIPTION
**Pull request by GlauberrBatista**
### TL;DR; 
When using redis-rb >=5, `resque-retry` crashes if the retry key for a job is `nil` since the `redis.get(key)` method does not support nil keys anymore. The fix is simply returning an empty string instead of `nil`.

---

### Long version

Delayed jobs that don't extend the `resque-retry` plugin don't have the `redis_retry_key` [method](https://github.com/lantins/resque-retry/blob/223897a7f920469543d96f3f335486d6d85e5036/lib/resque/plugins/retry.rb#L101) implemented, so the server will return `nil` when calling the `retry_key_for_job` [method](https://github.com/lantins/resque-retry/blob/223897a7f920469543d96f3f335486d6d85e5036/lib/resque-retry/server.rb#L41).

With redis-rb >=5, when we call `Resque.redis.get(value)` with the value being `nil`, it raises `TypeError: Unsupported command argument type: NilClass`, breaking the retry page on resque web.

The fix is very simple as we just need to return an empty string when the job does not have the retry key.

### CODE REVIEW

**Comment by jzaleski** [comment]
Thanks for the PR @GlauberrBatista -- Quick question, is it possible to modify this return value somewhere upstream as opposed to at this location?

**Comment by GlauberrBatista** [comment]
@jzaleski thank you for the response.

I don't think we can avoid touching that module. The `#retry_key_for_job` method is called only by its siblings `retry_attempts_for_job` and `cancel_retry`. But, if we are aiming to minimize the possible impact of the change, we could do something like `Resque.redis.get(retry_key_for_job(job).to_s)` on `retry_attempts_for_job` (the problematic method).

Should I go ahead and make that change?

**Comment by jzaleski** [comment]
@GlauberrBatista I made the changes to prevent this error case in this commit (just now): https://github.com/lantins/resque-retry/commit/c7fdeafb11a433a65891b386f67940d1ef094b40 -- it was largely inspired by your PR but a little different.

I also pushed a new version to `rubygems` here: https://rubygems.org/gems/resque-retry/versions/1.8.1

**Comment by GlauberrBatista** [comment]
@jzaleski thank you! 
I'm closing this

**Comment by jzaleski** [comment]
> @jzaleski thank you! 
> 
> I'm closing this

Very welcome. Cheers!
